src/services/refactors/generateGetAccessorAndSetAccessor.ts

```diff
@@ -0,0 +1,233 @@
+        return { renameFilename, renameLocation, edits };
+    }
+
+    function getUniqueName(baseName: string, fileText: string): string {
+        let nameText = baseName;
+        for (let i = 1; stringContains(fileText, nameText); i++) {
+            nameText = `${baseName}_${i}`;
+        }
+        return nameText;
+    }
+
+    function getRenameLocation(edits: ReadonlyArray<FileTextChanges>, renameFilename: string, fieldName: string, isDeclaredBeforeUse: boolean): number {
+        let delta = 0;
+        let lastPos = -1;
+        for (const { fileName, textChanges } of edits) {
+            Debug.assert(fileName === renameFilename);
+            for (const change of textChanges) {
+                const { span, newText } = change;
+                const index = newText.indexOf(fieldName);
+                if (index !== -1) {
+                    lastPos = span.start + delta + index;
+
+                    // If the reference comes first, return immediately.
+                    if (!isDeclaredBeforeUse) {
+                        return lastPos;
+                    }
+                }
+                delta += newText.length - span.length;
+            }
+        }
+
+        // If the declaration comes first, return the position of the last occurrence.
+        Debug.assert(isDeclaredBeforeUse);
+        Debug.assert(lastPos >= 0);
+        return lastPos;
+    }
+
+    function getAccessorModifiers(declaration: AccepedDeclaration, isStatic: boolean): NodeArray<Modifier> {
+        if (!declaration.modifiers || getModifierFlags(declaration) & ModifierFlags.Private) {
+            return createNodeArray(
+                append<Modifier>([createToken(SyntaxKind.PublicKeyword)],
+                    isStatic ? createToken(SyntaxKind.StaticKeyword) : undefined));
+        }
+        return declaration.modifiers;
+    }
+
+    function getFieldModifiers(isStatic: boolean): NodeArray<Modifier> {
+        return createNodeArray(
+            append<Modifier>([createToken(SyntaxKind.PrivateKeyword)],
+                isStatic ? createToken(SyntaxKind.StaticKeyword) : undefined));
+    }
+
+    function getPropertyDeclarationInfo(propertyDeclaration: PropertyDeclaration): DeclarationInfo | undefined {
+        if (!propertyDeclaration || propertyDeclaration.name.kind !== SyntaxKind.Identifier) return undefined;
+        if (!isClassLike(propertyDeclaration.parent) || !propertyDeclaration.parent.members) return undefined;
+
+        return {
+            isStatic: hasStaticModifier(propertyDeclaration),
+            classLikeContainer: propertyDeclaration.parent
+        };
+    }
+
+    function getParameterPropertyDeclarationInfo(parameterDeclaration: ParameterDeclaration): DeclarationInfo | undefined {
+        if (!parameterDeclaration || !isIdentifier(parameterDeclaration.name) || !isClassLike(parameterDeclaration.parent.parent) || !parameterDeclaration.parent.parent.members) return undefined;

```

Comment:
you already checked `!parameterDeclaration || !isIdentifier(parameterDeclaration.name) ` before